module4/homework/appendNewRecipe/AppendNewRecipe.cpp

```diff
@@ -0,0 +1,60 @@
+#include "AppendNewRecipe.hpp"
+
+#include <map>
+#include <fstream>
+
+const std::map<char, std::string> UNITS_NAMES{
+    {'g', "gram"},
+    {'s', "szklanka(i)"},
+    {'m', "mililitrow"}
+};
+
+RecipitList FormatIngredients(const IngredientsList &ingredients, const UnitsDeque &units){
+    if (ingredients.size() != units.size())
+        return {};
+
+    RecipitList result;
+    size_t counter{0};
+
+    for (auto &ingredient : ingredients)
+    {
+        auto unit = units[counter++];
+        auto text = std::to_string(unit.first) + ' ' + UNITS_NAMES.at(unit.second) + ' ' + ingredient;
+
+        result.push_back(text);
+    }
+
+    return result;
+}
+
+std::stringstream FormatRecipit(const StepList &steps, const IngredientsList &ingredients, const UnitsDeque &units){
+    std::stringstream output;
+
+    output << "Skladniki:" << '\n';
```

Comment:
You may store "Skladniki:" and "Kroki:" as const/constexpr, for example `ingredientsHeader` and `stepsHeader`.